mapflow/functional/layer_utils.py

```diff
@@ -528,11 +529,29 @@ def download_results_callback(self, response: QNetworkReply, processing) -> None
                 write_options
             )
         if error:
-            self.message_bar.pushWarning(self.tr("Error"),
-                                         self.tr('Failed to save results to file. '
-                                                 'Error code: {code}. Message: {message}').format(code=error,
-                                                                                                  message=msg))
-            return
+            self.message_bar.pushWarning(self.tr("Warning"),
+                                self.tr('Failed to save results to GeoPackage. '
+                                        'Error code: {code}. Message: {message}. ' 
+                                        'File will be saved as GeoJSON instead.').format(code=error, message=msg))
+            # Delete empty GeoPackage
+            Path(output_path).unlink()
+            # Save as GeoJSON instead
+            output_path = os.path.join(self.dlg.outputDirectory.text(), processing.id_)
```

Comment:
we can leverage pathlib here
`output_path = Path(output_path).with_suffix(".geojson")`

and for the name substitution
```
                while output_path.exists:
                    count += 1
                    output_path = output_path.with_stem(processing._id + f"_{count}"
              ```